java/src/main/java/com/saucelabs/saucebindings/SauceSession.java

```diff
@@ -34,54 +38,87 @@ public SauceSession(BaseConfigurations configs) {
 
     public SauceSession(SauceOptions options) {
         sauceOptions = options;
+        if (sauceOptions.visual() != null) {
+            dataCenter = DataCenter.VISUAL;
+            setVisual();
+        }
     }
 
     public RemoteWebDriver start() {
+        if (dataCenter.equals(DataCenter.VISUAL) && sauceOptions.visual() == null) {
+            sauceOptions.setVisualOptions(new VisualOptions());
+            setVisual();
+        }
+
+        if (isVisualSession && sauceOptions.sauce().getName() == null) {
```

Comment:
I wonder if this can be simplified by forcing the user to set test name in the constructor. I talk about it below...